### PR TITLE
Added angle field config

### PR DESCRIPTION
**Pull request by BeksOmega**
## The basics

- [X] I branched from develop
- [X] My pull request is against develop
- [X] My code follows the [style guide](https://developers.google.com/blockly/guides/modify/web/style-guide)

### Resolves

Work on #2722 

### Proposed Changes

* Adds config options: clockwise, offset, wrap, and round to the angle field.
* Fixes up a little thing with keyboard nav and wrapping.

### Reason for Changes

* Standardization of Configuration.
* Bugs be bad.

### Test Coverage

Added the following test blocks to the playground:
* Clockwise
* Offset 90
* Wrap 180
* Round 30
* No Round (0)
* Protractor Mode
* Compass Mode

For each of them I tested:
* Mouse Input
* Keyboard Nav Input
* Keyboard... key? Input

I also added unit tests for:
* Clockwise
* Offset
* Wrap
* Round
* Compass
* Protractor

Tested on:
* Desktop Chrome

### Documentation

Updates [here ](https://developers.google.com/blockly/guides/create-custom-blocks/fields/built-in-fields/angle#creation) and [here](https://developers.google.com/blockly/guides/create-custom-blocks/fields/built-in-fields/angle#customization).

Also [snapping ](https://developers.google.com/blockly/guides/create-custom-blocks/fields/built-in-fields/angle#snapping)needs a special note that a round value of 0 makes keyboard nav inopperable.

### Additional Information

N/A

### CODE REVIEW

File: core/field_angle.js [hunk 1]
```diff
@@ -42,12 +42,49 @@ goog.require('Blockly.utils.userAgent');
...
  * @extends {Blockly.FieldTextInput}
  * @constructor
  */
-Blockly.FieldAngle = function(opt_value, opt_validator) {
+Blockly.FieldAngle = function(opt_value, opt_validator, opt_config) {
+
+  /**
+   * Should the angle increase as the angle picker is moved clockwise (true)
+   * or counterclockwise (false)
+   * @see CLOCKWISE
```
Comment (samelhusseini) [inline core/field_angle.js]: I think the @see, needs to be the fully qualified name: ``Blockly.FieldAngle.CLOCKWISE``

File: core/field_angle.js [hunk 2]
```diff
@@ -375,17 +438,22 @@ Blockly.FieldAngle.prototype.onHtmlInputKeyDown_ = function(e) {
...
+  var value = Number(opt_newValue);
+  if (isNaN(value) || !isFinite(value)) {
     return null;
   }
-  if (n < 0) {
-    n += 360;
+  return this.wrapValue_(value);
+};
+
+Blockly.FieldAngle.prototype.wrapValue_ = function(value) {
```
Comment (samelhusseini) [inline core/field_angle.js]: jsdoc

File: core/field_angle.js [hunk 3]
```diff
@@ -70,51 +107,87 @@ Blockly.FieldAngle.fromJson = function(options) {
...
 
 /**
- * Radius of protractor circle.  Slightly smaller than protractor size since
- * otherwise SVG crops off half the border at the edges.
+ * The default amount to round angles to when using a mouse or keyboard nav
+ * input. Must be a positive integer to support keyboard navigation.
+ * @const {number}
  */
-Blockly.FieldAngle.RADIUS = Blockly.FieldAngle.HALF - 1;
+Blockly.FieldAngle.ROUND = 15;
```
Comment (samelhusseini) [inline core/field_angle.js]: Why the re-order?

Comment (BeksOmega) [reply core/field_angle.js]: It feels better.
Half, Radius, Clockwise, Offset, Wrap, Round vs
Round, Half, Clockwise, Offset, Wrap, Radius

I didn't think about it causing merge conflicts though, so I can change it back.

**Comment by samelhusseini** [comment]
Thanks @BeksOmega!

**Pull request merged**